### PR TITLE
sc-im: aligning a selection of cells cause crash

### LINKED ISSUE #335130: sc-im: aligning a selection of cell's cause crash
### Describe the bug

Sc-im crashes with buffer overflow when trying ot align a visual selection.

```
*** buffer overflow detected ***: terminated
[1]    1081326 IOT instruction (core dumped)  sc-im
```

### Steps To Reproduce
1. enter text in A1 (`< blabla`)
2. enter text in A2 (`<blabla`)
3. visual select both cell's
4. use `f + }` to align right.

### Expected behavior
both cells should align right.

### Screenshots

https://github.com/user-attachments/assets/269a93cd-83f8-4dcf-9923-02b170f6ce06

### Additional context
The upstream issue is here: https://github.com/andmarti1424/sc-im/issues/884

### Notify maintainers
@dotlambda 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.6.32, NixOS, 24.05 (Uakari), 24.05.20240531.805a384`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/i1aw9jjgxcvyd642s12kw3iasmarwd42-source`output here
```
---

### PR DESCRIPTION
**Pull request by mipmip**
## Description of changes

By disabling fortify in gcc this fixes #335130 `sc-im: aligning a selection of cell's cause crash`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dotlambda** [comment]
> * [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Not true. The commit message should start with `sc-im`.

**Comment by smancill** [comment]
Please edit the commit summary to `sc-im: fix crash` and put more details in the body.

File: pkgs/applications/misc/sc-im/default.nix [hunk 1]
```diff
@@ -46,6 +46,8 @@ stdenv.mkDerivation rec {
 
   makeFlags = [ "prefix=${placeholder "out"}" ];
 
+  hardeningDisable = [ "fortify" ];
```
Comment (dotlambda) [inline pkgs/applications/misc/sc-im/default.nix:49]:
```suggestion
  # https://github.com/andmarti1424/sc-im/issues/884
  hardeningDisable = [ "fortify" ];
```

**Comment by mipmip** [comment]
> Please edit the commit summary to `sc-im: fix crash` and put more details in the body.

I just did. Sorry for my beginner mistakes.

**Comment by smancill** [comment]
Looks OK to me but I cannot reproduce the bug on Darwin. Needs someone with Linux to test it.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 335133` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sc-im</li>
  </ul>
</details>

**Comment by korrat** [comment]
I was able to verify the commit on x86_64-linux.

On recent nixpkgs (`9g23szywla53rajzivcisxkiv415hwy0-sc-im-0.8.4`), sc-im dumps core when trying to align the cells. With this PR (`x6psi9sw54xyq2aa34k0p3ss1pxb8c4i-sc-im-0.8.4`), aligning works as expected.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 335133`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sc-im</li>
  </ul>
</details>

**Comment by smancill** [comment]
@mipmip please rebase to handle merge conflicts.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by mipmip** [comment]
Thanks, I'm happy as this is my first PR 🎉

**Pull request merged**